### PR TITLE
Fixed the build warning when comliling with gcc 7.4.0

### PR DESCRIPTION
**Pull request by sherman-the-tank**
The erroes are caused by

-Werror=sign-compare
-Werror=suggest-override

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -195,6 +193,9 @@ add_compile_options(-Wall)
 add_compile_options(-Werror)
 add_compile_options(-Wunused-parameter)
 add_compile_options(-Wshadow)
+# Disable the complain anout comparison between signed and unsigned integers
+# TODO (sye) Need to figure out a better solution
+add_compile_options(-Wno-error=sign-compare)
```
Comment (monadbobo) [inline CMakeLists.txt]: We should decide whether to add this option based on the compiler version.

Comment (sherman-the-tank) [reply CMakeLists.txt]: I prefer not to add this option at the global level at all. We can fix all warnings in our code. The trouble is caused by folly. This is just a temporary solution. That's why I added a TODO here

**Comment by sherman-the-tank** [comment]
Go jenkins

**Comment by sherman-the-tank** [comment]
Jenkins go

**Comment by dutor** [comment]
Closed for now.
If this issue still exists, please reopen it.

**Pull request closed without merging**